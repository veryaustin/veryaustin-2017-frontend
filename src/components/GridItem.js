import React, { PropTypes } from "react";
import styled from "styled-components";

// Container for individual grid item
const Item = styled.div`
  width: 50vw;
  height: 50vw;
  background-color: ${props => props.color};
  transition: all 0.1s ease-in-out;
  &:hover {
    transform: scale(1.02);
    box-shadow: 0px 0px 10px #000000;
  }
  @media only screen and (min-width: 769px) {
    width: 20vw;
    height: 20vw;
  }
`;

// Logo image for an individual grid item
const ItemImage = styled.img`
  width: 50%;
  height: 50%;
  margin: 25%;
`;

// GridItem Component
const GridItem = props => {
  const { site_url, background_color, logo_url } = props.gridItem;
  return (
    <a href={site_url} target="_blank">
      <Item color={background_color}>
        <ItemImage src={logo_url} />
      </Item>
    </a>
  );
};

// GridItem Props Validation
GridItem.propTypes = {
  gridItem: PropTypes.object
};

export default GridItem;
