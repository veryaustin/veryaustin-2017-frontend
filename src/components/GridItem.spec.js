import React from "react";
import { shallow } from "enzyme";
import GridItem from "./GridItem";
import toJson from "enzyme-to-json";

describe("GridItem Component", () => {
  // Component Tests
  let wrapper;
  beforeEach(() => {
    const sampleItem = {
      site_url: "http://example.com",
      background_color: "#119f4d",
      logo_url: "http://example.com/test.svg"
    };
    wrapper = shallow(<GridItem gridItem={sampleItem} />);
  });

  it("renders the gridItem with site link, background color & logo", () => {
    const gridItem = wrapper.find("styled.div");
    const link = wrapper.find("a[href='http://example.com/']");
    expect(gridItem).toExist;
    expect(link).toExist;
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
