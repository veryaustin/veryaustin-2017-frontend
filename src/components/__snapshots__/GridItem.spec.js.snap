// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridItem Component renders the gridItem with site link, background color & logo 1`] = `
<a
  href="http://example.com"
  target="_blank"
>
  <styled.div
    color="#119f4d"
  >
    <styled.img
      src="http://example.com/test.svg"
    />
  </styled.div>
</a>
`;
